### PR TITLE
Rewrite Docstrings + Change Netlify build to branch from Pip

### PR DESCRIPTION
**Pull request by NirantK**

### CODE REVIEW

**Comment by NirantK** [comment]
BEFORE: 
image

AFTER: 
image

File: tools/generate_docs_netlify.sh [hunk 1]
```diff
@@ -7,10 +7,13 @@ cd "$(dirname "$0")/../"
 
 brew install pandoc
 
-pip install qdrant-client
+curl -sSL https://install.python-poetry.org | python3 -
+export PATH="/opt/buildhome/.local/bin:$PATH"
+poetry install
 
 pip install sphinx==4.5.0
+pip install ipython
```
Comment (joein) [inline tools/generate_docs_netlify.sh:15]: why isn't it in pyproject.toml dev group?

Comment (NirantK) [reply tools/generate_docs_netlify.sh:15]: This is needed only for the docs build. Nothing else. Should we include that in the dev group? If yes, will add this, Sphinx, pygments and other docs dependencies there.

Comment (joein) [reply tools/generate_docs_netlify.sh:15]: they are already there

Comment (NirantK) [reply tools/generate_docs_netlify.sh:15]: Will add these to a new `docs` group in a different PR

**Review by joein — APPROVED** [review]

**Pull request merged**